kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/SchemaUtils.java

```diff
@@ -350,7 +352,7 @@ static SchemaChanges computeSchemaChangesById(
       }
     }
 
-    for (Map.Entry<Long, StructField> entry : currentFieldIdToField.entrySet()) {
+    for (Map.Entry<Integer, StructField> entry : currentFieldIdToField.entrySet()) {
```

Comment:
I cleaned up all the use of `long` for field IDs, they are always 32 bit integers.